pkg/destroy/openstack/openstack.go

```diff
@@ -655,27 +657,37 @@ func deleteContainers(opts *clientconfig.ClientOpts, filter Filter, logger logru
 			// Openshiftclusterid in the X-Container-Meta- HEAD output
 			titlekey := strings.Title(strings.ToLower(key))
 			if metadata[titlekey] == val {
-				listOpts := objects.ListOpts{Full: false}
-				allPages, err := objects.List(conn, container, listOpts).AllPages()
-				if err != nil {
-					logger.Error(err)
-					return false, nil
-				}
-				allObjects, err := objects.ExtractNames(allPages)
-				if err != nil {
-					logger.Error(err)
-					return false, nil
-				}
-				for _, object := range allObjects {
-					logger.Debugf("Deleting object %q", object)
-					_, err = objects.Delete(conn, container, object, nil).Extract()
+				logger.Debugf("Bulk deleting container %q objects", container)
+				pager := objects.List(conn, container, &objects.ListOpts{
+					Limit: 50,
+				})
+				err = pager.EachPage(func(page pagination.Page) (bool, error) {
+					objectsOnPage, err := objects.ExtractNames(page)
 					if err != nil {
-						// Ignore the error if the object cannot be found and return with an appropriate message if it's another type of error
-						if _, ok := err.(gophercloud.ErrDefault404); !ok {
-							logger.Errorf("Removing object %q from container %q failed: %v", object, container, err)
-							return false, nil
+						return false, err
+					}
+					resp, err := objects.BulkDelete(conn, container, objectsOnPage).Extract()
+					if err != nil {
+						return false, err
+					}
+					if len(resp.Errors) > 0 {
+						// Convert resp.Errors to golang errors.
+						// Each error is represented by a list of 2 strings, where the first one
+						// is the object name, and the second one contains an error message.
+						errs := make([]error, len(resp.Errors))
+						for _, objectError := range resp.Errors {
+							errs = append(errs, errors.Errorf("cannot delete object %v: %v", objectError[0], objectError[1]))
```

Comment:
If `resp.Errors` has 5 elements, here "errs" is a slice with length 5; appending one more will result in "errs" having length 6 ([here](https://play.golang.org/) an example).

What you can do instead, is directly assigning by index:

```Go
for i, objectError := range resp.Errors {
	errs[i] = errors.Errorf("cannot delete object %v: %v", objectError[0], objectError[1])
}
```

Reply:
sure! thanks for noticing this